win32ss/user/ntuser/window.c

```diff
@@ -2980,8 +2980,9 @@ BOOLEAN co_UserDestroyWindow(PVOID Object)
    IntSendDestroyMsg(UserHMGetHandle(Window));
 
    // Destroy the default IME window if necessary
-   if (IS_IMM_MODE() && !(ti->TIF_flags & TIF_INCLEANUP) &&
-       ti->spwndDefaultIme && !IS_WND_IMELIKE(Window) && !(Window->state & WNDS_DESTROYED))
+   if (IS_IMM_MODE() && (ti != NULL) && !(ti->TIF_flags & TIF_INCLEANUP) &&
+       ti->spwndDefaultIme && ti->spwndDefaultIme != Window &&
+       !(Window->state & WNDS_DESTROYED) && !IS_WND_IMELIKE(Window))
```

Comment:
I don't want to say it would be worse now than before. But curious: What made you change your initial approach to simply move the code upwards?

Reply:
The cause of BSoD is in `IS_WND_IMELIKE`.
The `WNDS_DESTROYED` window can't accept `IS_WND_IMELIKE`.